### PR TITLE
Rule 'no-anonymous-observer' is deprecated. Removed from recommended,…

### LINKED ISSUE #3422: Observer HOC is not getting displayName from forwardRef
When i wrap forwardRef by observer, any infomation that can be used as display name is lost.

For example:

```
const ComponentWithForwardRef = forwardRef(function ComponentWithForwardRef(
  {},
  ref
) {
  return <div ref={ref} />;
});

const ObserverComponent = observer(ComponentWithForwardRef)
```
So ObserverComponent has name **ObserverForwardRef** instead **ComponentWithForwardRef**.

If i explicitly set
```
ComponentWithForwardRef.displayName = "ComponentWithForwardRef"
const ObserverComponent = observer(ComponentWithForwardRef)
```
i see same thing

The only variant ive found is to explicitly set displayName after wrapping in observer
```
const ObserverComponent = observer(ComponentWithForwardRef)
ObserverComponent .displayName = "ComponentWithForwardRef"
```
but it isnt look pretty

Also, mobx/no-anonymous-observer is not happy with all of listed variants.

Any ideas?

Checked for mobx-react-lite: 3.4.0
Here: https://codesandbox.io/s/condescending-glade-yx3qgy?file=/index.js

### PR DESCRIPTION
**Pull request by kazah96**
resolves #3422

### CODE REVIEW

File: packages/eslint-plugin-mobx/README.md [hunk 1]
```diff
@@ -82,9 +82,23 @@ It's a bit opinionated and can lead to a lot of false positives depending on you
...
 
-### mobx/no-anonymous-observer
+### mobx/no-anonymous-observer (deprecated)
 
 Forbids anonymous functions or classes as `observer` components.
 Improves debugging experience and [avoids problem with inability to customize `displayName`](https://github.com/mobxjs/mobx/issues/2721).
 Plays nice with `eslint-plugin-react-hooks` and `mobx/missing-observer` as both of these don't not recognize anonymous function as component.
 **Autofix** infers the name from variable if possible.
+
+_Note: Due to known [issues](https://github.com/mobxjs/mobx/issues/3422), it ts preferably to use [react/display-name](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/display-name.md) with [componentWrapperFunctions](https://github.com/jsx-eslint/eslint-plugin-react). So your .eslintrc config could look like this:_
```
Comment (urugator) [inline packages/eslint-plugin-mobx/README.md:92]: Please move before the original description and replace with:

 ___Deprecated__ in favor of [react/display-name](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/display-name.md) + [`componentWrapperFunctions`](https://github.com/jsx-eslint/eslint-plugin-react).  Example of __`.eslintrc`__:_
```json
{
  "rules": {
    "react/display-name": "warn"
  },
  "settings": {
    "componentWrapperFunctions": [
      "observer"
    ]
  }
}
```
---

Forbids anonymous functions or class...
...
...

Comment (kazah96) [reply packages/eslint-plugin-mobx/README.md]: @urugator fixed.

**Comment by urugator** [comment]
Thank you!

**Comment by iChenLei** [comment]
`No Changeset found`, so that's why recent release #3428 not include `eslint-plugin-mobx` @urugator

**Comment by urugator** [comment]
Done, released as eslint-plugin-mobx@0.0.9

**Pull request merged**